riscv/processor.h

```diff
@@ -209,8 +209,17 @@ class processor_t : public abstract_device_t
     if (ext >= 'A' && ext <= 'Z')
       return state.misa->extension_enabled(ext);
     else
-      return isa->extension_enabled(ext);
+      return isa->extension_enabled(ext) && !extension_disable_table[ext];
   }
+
+  void enable_extension(unsigned char ext) {
+    extension_disable_table[ext] = false;
+  }
+
+  void disable_extension(unsigned char ext) {
+    extension_disable_table[ext] = true;
+  }
```

Comment:
Please move these below (or above) `extension_enabled_const()` so we can keep that next to its kin `extension_enabled()`.

Also, `extension_enabled_const()` is now subtly broken. See the comment about "assume this can't change" which is no longer a valid assumption.

There needs to be somewhere a list of extensions that are allowed to be dynamically disabled. `disable_extension()` (and the `set_extension_enable()` I suggested) should check that list and `abort()` if any code tries to disable something not in that list. Then `extension_enabled_const()` can check that same list. Only then can assumptions about unchanging extensions be validated, and caught immediately if that should ever change.

Reply:
OK. I'll update it.